core/test/element_arithmetic_test.cpp

```diff
@@ -243,3 +244,33 @@ TEST_F(ElementNanArithmeticTest, add_equals_with_rhs_int_lhs_int) {
   nan_add_equals(lhs, 2);
   EXPECT_EQ(3, lhs);
 }
+
+class ElementMatrixArithmeticTest : public ::testing::Test {
```

Comment:
No need to add (detailed) tests? This is just a template with no specialization? If anything, we need to add tests for special cases such as `x *= x` to ensure aliasing is not an issue?

Reply:
Ok I agree we need to test the special cases `x *= x` but i'm not clear on your comment about not needing to add detailed tests for this.

> I feel this is really odd, given that for scipp the "matrix" dtype really means "rotation"

Aside from matrix creation via `from_rotvec` which does for example perform unit checks on the input vectors, I'm not aware on any implementation of the current `matrix` that makes it only suitable for rotations. Nevertheless, I'm happy to drop the `+=` `-=` support as it makes things difficult with `nan_sum` reduction operations in any case.

Reply:
> I'm not aware on any implementation of the current matrix that makes it only suitable for rotations.

Hmm, right, I think I meant a transformation matrix. But would that make addition useful?

Reply:
I suppose this links into the wider discussion on whether we need more types for transformation, e.g., affine transformations.

Reply:
Yes affine transformations would be really useful. The might have good application in the nexus loaders amongst other places.

Reply:
> Ok I agree we need to test the special cases x *= x but i'm not clear on your comment about not needing to add detailed tests for this.

Well, the question is what you are trying to test? We don't generally test our kernel for all supported types, and unless there is a template specialization I feel that is indeed not necessary.

So the question becomes: Are you trying to test that `*=` does what you expect for `Eigen::Matrix3d`? Then you should consider adding a test for that [our current linear algebra library], but not the templated kernel?